src/dotnet/Chat.UI.Blazor/Services/RightPanelUI.cs

```diff
@@ -5,15 +5,53 @@ namespace ActualChat.Chat.UI.Blazor.Services;
 
 public class RightPanelUI
 {
-    public IStoredState<bool> IsVisible { get; }
+    private readonly IMutableState<bool> _isVisible;
+
+    private HistoryUI HistoryUI { get; }
+    private BrowserInfo BrowserInfo { get; }
+
+    public IState<bool> IsVisible
+        => _isVisible;
 
     public RightPanelUI(IServiceProvider services)
     {
+        HistoryUI = services.GetRequiredService<HistoryUI>();
+        BrowserInfo = services.GetRequiredService<BrowserInfo>();
+
         var stateFactory = services.StateFactory();
-        var localSettings = services.GetRequiredService<LocalSettings>().WithPrefix(nameof(RightPanelUI));
-        IsVisible = stateFactory.NewKvasStored<bool>(
-            new (localSettings, nameof(IsVisible)) {
-                InitialValue = false,
-            });
+        // TODO: On Desktop we can switch between 2 wide and narrow layout. Decide how to better handle this?
+        if (BrowserInfo.ScreenSize.Value.IsNarrow())
+            _isVisible = stateFactory.NewMutable<bool>();
+        else {
+            var localSettings = services.GetRequiredService<LocalSettings>().WithPrefix(nameof(RightPanelUI));
+            _isVisible = stateFactory.NewKvasStored<bool>(
+                new (localSettings, nameof(IsVisible)) {
+                    InitialValue = false,
+                });
+        }
     }
+
+    public void Switch()
+        => ChangeVisibility(!IsVisible.Value);
+
+    public void ChangeVisibility(bool visible)
+    {
+        if (_isVisible.Value == visible)
+            return;
+
+        var screenSize = BrowserInfo.ScreenSize.Value;
+        if (screenSize.IsNarrow()) {
+            if (visible)
+                HistoryUI.NavigateTo(
+                    () => InnerChangeVisibility(true),
+                    () => InnerChangeVisibility(false));
+            else
+                _ = HistoryUI.GoBack();
+        }
+        else
+            InnerChangeVisibility(visible);
+    }
+
+    private void InnerChangeVisibility(bool visible)
```

Comment:
what does inner mean here? like Internal? Don't we use internal suffix for such cases?